src/resultSet.js

```diff
@@ -1,46 +1,219 @@
 const Statistics = require("./statistics"),
-	Record = require("./record");
+    Record = require("./record");
+Node = require("./node");
+Edge = require("./edge");
+
+const ResultSetColumnTypes = {
+    COLUMN_UNKNOWN: 0,
+    COLUMN_SCALAR: 1,
+    COLUMN_NODE: 2,
+    COLUMN_RELATION: 3
+}
+
+const ResultSetScalarTypes = {
+    PROPERTY_UNKNOWN: 0,
+    PROPERTY_NULL: 1,
+    PROPERTY_STRING: 2,
+    PROPERTY_INTEGER: 3,
+    PROPERTY_BOOLEAN: 4,
+    PROPERTY_DOUBLE: 5
+}
 
 /**
  * Hold a query result
  */
 class ResultSet {
-	constructor(resp) {
-		this._position = 0;
-		this._statistics = new Statistics(resp[1]);
-
-		let result = resp[0];
-
-		// Empty result set
-		if (result === null || result.length === 0) {
-			this._header = [];
-			this._totalResults = 0;
-			this._results = [];
-		} else {
-			this._header = result[0];
-			this._totalResults = result.length - 1;
-			this._results = new Array(this._totalResults);
-			for (let i = 0; i < this._totalResults; ++i) {
-				this._results[i] = new Record(this._header, result[i + 1]);
-			}
-		}
-	}
-
-	getHeader() {
-		return this._header;
-	}
-
-	hasNext() {
-		return this._position < this._totalResults;
-	}
-
-	next() {
-		return this._results[this._position++];
-	}
-
-	getStatistics() {
-		return this._statistics;
-	}
+    constructor(graph) {
+        //_graph is graph api
+        this._graph = graph;
```

Comment:
Comment at the same line as the prop,
this._graph = graph;    //_graph is graph api